pynest/examples/pynest-microcircuit/network.py

```diff
@@ -0,0 +1,347 @@
+                self.voltmeter.append(volt)
+
+        if 'spike_detector' in self.net_dict['rec_dev']:
+            print 'spike_detectors created'
+        if 'voltmeter' in self.net_dict['rec_dev']:
+            print 'voltmeter created'
+
+    # thalamic neurons are created here
+    def create_thalamic_input(self):
+        if self.stim_dict['thalamic_input']:
+            print 'thalamic input provided'
+            self.thalamic_population = nest.Create(
+                                           'parrot_neuron',
+                                           self.stim_dict['n_thal'])
+            self.thalamic_weight = get_weight(self.stim_dict['PSP_th'])
+            self.stop_th = (self.stim_dict['th_start'] +
+                            self.stim_dict['th_duration'])
+            self.poisson_th = nest.Create('poisson_generator')
+            nest.SetStatus(
+                self.poisson_th, {'rate': self.stim_dict['th_rate'],
+                                  'start': self.stim_dict['th_start'],
+                                  'stop': self.stop_th})
+            nest.Connect(self.poisson_th, self.thalamic_population)
+            self.nr_synapses_th = synapses_th_matrix()
+            if self.K_scl != 1:
+                self.thalamic_weight = self.thalamic_weight/(self.K_scl**0.5)
+                self.nr_synapses_th = (self.nr_synapses_th * self.K_scl)
+        else:
+            print 'thalamic input not provided'
+
+    # poisson generators are created in this function
+    def create_poisson(self):
+        if self.net_dict['poisson_input']:
+            rate_ext = self.net_dict['bg_rate'] * self.K_ext
+            self.poisson = []
+            for i, target_pop in enumerate(self.pops):
+                    poisson = nest.Create('poisson_generator')
+                    nest.SetStatus(poisson, {'rate': rate_ext[i]})
+                    self.poisson.append(poisson)
+            print 'Poisson background input created'
+
+    # dc generators are created in this functions
+    def create_dc_generator(self):
+        if self.stim_dict['dc_input']:
+            dc_amp_stim = self.net_dict['K_ext'] * self.stim_dict['dc_amp']
+            self.dc = []
+            print 'DC_amp_stim', dc_amp_stim
+            for i, target_pop in enumerate(self.pops):
+                dc = nest.Create('dc_generator',
+                                 params={'amplitude': dc_amp_stim[i],
+                                         'start': self.stim_dict['dc_start'],
+                                         'stop': self.stim_dict['dc_start'] +
+                                         self.stim_dict['dc_dur']})
+                self.dc.append(dc)
+            print 'DC generator created'
+
+    # connections between neuronal populations are created in this function
+    def create_connections(self):
+        mean_delays = self.net_dict['mean_delay_matrix']
+        std_delays = self.net_dict['std_delay_matrix']
+        norm_clp = 'normal_clipped'
```

Comment:
This variable definition seems obsolete. You can just directly insert 'normal_clipped' in all the dictionaries below.